pkg/controller/backup_sync_controller_test.go

```diff
@@ -154,3 +158,133 @@ func TestBackupSyncControllerRun(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteUnused(t *testing.T) {
+	tests := []struct {
+		name               string
+		getAllBackupsError error
+		cloudBackups       []*v1.Backup
+		k8sBackups         []*arktest.TestBackup
+		namespace          string
+		numMacthingBackups int
+	}{
+		{
+			name:      "no overlapping backups",
+			namespace: "ns-1",
+			cloudBackups: []*v1.Backup{
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-1").Backup,
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-2").Backup,
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backup-3").Backup,
+			},
+			k8sBackups: []*arktest.TestBackup{
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupA").WithPhase(v1.BackupPhaseCompleted),
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupB").WithPhase(v1.BackupPhaseCompleted),
+				arktest.NewTestBackup().WithNamespace("ns-1").WithName("backupC").WithPhase(v1.BackupPhaseCompleted),
+			},
+			numMacthingBackups: 0,
```

Comment:
I think this should be 3.

None of the 3 `k8sBackups` entries are present in `cloudBackups`. Therefore, they should be deleted.

Reply:
Hm, no, this is correct. But something may not be mocked correctly since the `k8sBackups` still show up in the `c.backupLister.Backups(c.namespace).List` call.